extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -165,7 +175,28 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 
 		try {
 			let pythonExists = await utils.exists(this._pythonExecutable);
-			if (!pythonExists || forceInstall) {
+			let upgradePython = false;
+			// Warn users that some packages may need to be reinstalled after updating Python versions
+			if (pythonExists && !this._usingExistingPython && await this.getInstalledPythonVersion(this._pythonExecutable) !== constants.pythonVersion) {
```

Comment:
Are we targeting just Python 3.8, or do want them to be _at least_ on Python 3.8? 3.9 is the current version, so just doing an equality comparison here would prompt for a version downgrade if they're using that.

Reply:
Agreed, should do a semver compare here

Reply:
Since I am also checking `usingExistingPython`, users who are using their own Python 3.9 would not be prompted. Although I can change the comparison to at least, just in case someone unchecked `usingExistingPython` setting even though they are using their own Python.